src/symlink.rs

```diff
@@ -0,0 +1,34 @@
+use std::path::Path;
+
+#[cfg(unix)]
+pub(crate) use std::os::unix::fs::symlink as symlink_file;
+
+/// Creates a symbolic link to a file.
+///
+/// On Windows, creating a symbolic link to a file requires Administrator
+/// rights. Fall back to copying if creating the symbolic link fails.
+#[cfg(windows)]
+pub(crate) fn symlink_file<P: AsRef<Path>, Q: AsRef<Path>>(
+    original: P,
+    link: Q,
+) -> std::io::Result<()> {
+    use std::sync::atomic::{AtomicBool, Ordering::SeqCst};
+
+    static HAVE_PRINTED_WARNING: AtomicBool = AtomicBool::new(false);

```

Comment:
An AtomicBool shouldn't be necessary because the producer is single threaded, but better to be safe than sorry in case we change that in the future.